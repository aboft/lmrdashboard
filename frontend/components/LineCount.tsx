export default function LineCount(props: ILines) {
    return <div>[{ props.dateCreated && new Date(props.dateCreated).toDateString()}]: {props.count}{props.message}</div>;
}

export interface ILines {
    count?: number
    dateCreated?: string
    message?: string
}

