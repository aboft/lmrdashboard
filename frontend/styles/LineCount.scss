.linecount-header {
  font-weight: bold;
  font-size: 1em;
  margin-bottom: 0;

  .linecount-tooltip {
    font-weight: normal !important;
    opacity: 50%;
  }
}
