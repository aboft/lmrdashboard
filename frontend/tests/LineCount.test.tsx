import React from 'react';
import renderer from 'react-test-renderer';
import { expect } from '@jest/globals';
import LineCount from '../components/LineCount';

test('LineCount Component has <10% bot messages', () => {
  const component = renderer.create(
    <LineCount key={'key'} lineCount={1000} botLines={55} date={'2021-01-01'} message={'foo'} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LineCount Component has 25% bot messages', () => {
  const component = renderer.create(
    <LineCount key={'key'} lineCount={100} botLines={25} date={'2021-01-01'} message={'foo'} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LineCount Component has 31% bot messages', () => {
  const component = renderer.create(
    <LineCount key={'key'} lineCount={1337} botLines={420} date={'2021-01-01'} message={'foo'} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LineCount Component has 50% bot messages', () => {
  const component = renderer.create(
    <LineCount key={'key'} lineCount={24} botLines={12} date={'2021-01-01'} message={'foo'} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LineCount Component has 100% bot messages', () => {
  const component = renderer.create(
    <LineCount key={'key'} lineCount={24} botLines={24} date={'2021-01-01'} message={'foo'} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
