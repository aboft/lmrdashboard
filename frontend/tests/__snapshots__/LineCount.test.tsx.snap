// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineCount Component has 25% bot messages 1`] = `
<div>
  2021-01-01
  : 
  100
   (
  25%
  )
  foo
</div>
`;

exports[`LineCount Component has 31% bot messages 1`] = `
<div>
  2021-01-01
  : 
  1337
   (
  31%
  )
  foo
</div>
`;

exports[`LineCount Component has 50% bot messages 1`] = `
<div>
  2021-01-01
  : 
  24
   (
  50%
  )
  foo
</div>
`;

exports[`LineCount Component has 100% bot messages 1`] = `
<div>
  2021-01-01
  : 
  24
   (
  100%
  )
  foo
</div>
`;
